src/components/Dashboard.tsx

```diff
@@ -94,38 +85,89 @@ const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template })
   const templateType = template.code
-  const { loading, applicationCount } = useListApplications({
-    type: templateType,
-    perPage: 1,
-    ...filter.query,
-  })
+  const [loadedFiltersCount, setLoadedFiltersCount] = useState(0)
+  const [totalMatchFilter, setTotalMatchFilter] = useState<{ [key: number]: number }>(
+    filters.reduce((totalPerFilter, filter) => ({ ...totalPerFilter, [filter.id]: 0 }), {})
+  )
+
+  const arrayFilters = filters.reduce((arrayFilters: { [key: string]: string }[], element) => {
+    if (typeof element.query === 'object') {
+      const queryObj = element.query as { [key: string]: string }
+      arrayFilters.push(queryObj)
+    }
+    return arrayFilters
+  }, [])
+
+  const queryMultiFilters = {
+    templateCode: { equalToInsensitive: templateType },
+    ...constructOrObjectFilters(arrayFilters),
+  }
+
+  const { loading, applications } = useListApplications(
+    {}, // Passing empty to use already construct GraphQL query on queryMultiFilters
+    queryMultiFilters
+  )
 
   useEffect(() => {
-    if (!loading) setLoadingFilters((currentCount: number) => currentCount + 1)
-  }, [loading])
+    if (applications) {
+      filters.forEach(({ id, query }) => {
+        if (typeof query === 'object') {
+          const queryObj = query as { [key: string]: string }
+          const filteredApplications = Object.entries(applications).filter(([_, application]) => {
+            // Each filter is currently delimited to a single check!
+            const key = Object.keys(queryObj)[0]
+            const value = Object.values(queryObj)[0]
 
-  const applicationListUserRole =
+            switch (key) {
+              case 'outcome':
+                return application.outcome === value
+              case 'status':
+                return application.status === value
+              case 'reviewerAction':
+                return application.reviewerAction === value
+              case 'assignerAction':
+                return application.assignerAction === value
+              default:
+                return false
+            }
+          })
+          totalMatchFilter[id] = filteredApplications.length
+        }
+        setTotalMatchFilter(totalMatchFilter)
+        setLoadedFiltersCount((currentCount: number) => currentCount + 1)
+      })
+    }
+  }, [applications])
+
+  const userRole = (filter: Filter) =>
     filter.userRole === PermissionPolicyType.Apply ? USER_ROLES.APPLICANT : USER_ROLES.REVIEWER
 
-  const constructLink = () =>
-    `/applications?type=${templateType}&user-role=${applicationListUserRole}&${Object.entries(
-      filter.query
-    )
+  const constructLink = (filter: Filter) =>
+    `/applications?type=${templateType}&user-role=${userRole(filter)}&${Object.entries(filter.query)
       .map(([key, value]) => `${key}=${value}`)
       .join('&')}`
 
-  if (applicationCount === 0) return null
+  if (loading || loadedFiltersCount < filters.length) return <LoadingSmall />
```

Comment:
Yeah, as above -- if you get rid of "loadedFiltersCount", then `loading` is all you need to check.